cmds/copy.c

```diff
@@ -27,9 +27,10 @@ static void cmd_copyInfo(void)
 }
 
 
-static int cmd_cpphfs2phfs(handler_t srcHandler, addr_t srcAddr, size_t srcSz, handler_t dstHandler, addr_t dstAddr, size_t dstSz)
+static ssize_t cmd_cpphfs2phfs(handler_t srcHandler, addr_t srcAddr, size_t srcSz, handler_t dstHandler, addr_t dstAddr, size_t dstSz)
 {
 	int res;
+	addr_t addr = dstAddr;
```

Comment:
You have 3 variables that you modify on each pass of the loop. Maybe using one variable (which would represent current position in buffers), adding this value to addresses and comparing it with const size would be better? You wouldn't need to calculate retval this way and IMHO code would be easier to read and less prone to errors.

Reply:
Yeah, it can be simplified. However, the `res` value might differ from read and write value, it is the worst scenario. 
For this reason I decided to use more than one variable.

Reply:
Yeah, good point. What if res is different for read and write? E.g. we read 20 bytes and manage to write only 10. Wouldn't we silently drop 10 bytes of data? I think that write should have it's own loop to write all read data before next read operation.

Reply:
Or we can read again from offset we got from write - either way, one indexing variable is better ;)